pkg/sql/conn_executor_test.go

```diff
@@ -924,3 +924,117 @@ func (f *dynamicRequestFilter) filter(
 func noopRequestFilter(ctx context.Context, request roachpb.BatchRequest) *roachpb.Error {
 	return nil
 }
+
+func TestBigClientMessage(t *testing.T) {
```

Comment:
see my comment below: I believe this test is best served by datadriven testing via pgwire_test.go

Reply:
Going to push back on this one -- this introduces a lot of cruft with pg wire tests, e.g.:
* needing to setup a new connection for the cluster setting to kick in
* needing to large strings with replace directives
* needed to differentiate hard errors which involve a connection termination (i.e. io.EOF), as opposed to ErrorResponse with retries (as from experience, CLUSTER SETTING in tests may not immediately kick in).
* big messages are unwieldly to commit as an expected result (we currently expect at least 16kB as the shortest buffer size...).
* it does not integration test well with an driver

I've added some more cases here to test through.